checker/tests/run-pass/array_index.rs

```diff
@@ -12,12 +12,11 @@ extern crate mirai_annotations;
 
 pub fn foo(arr: &mut [i32], i: usize) {
     arr[i] = 123; //~ possible array index out of bounds
-    // If we get here i is known to be within bounds, so no warning below.
-    bar(arr, i);
+    bar(arr, i); //~ possible array index out of bounds
```

Comment:
I take it that it's intentional that you get the warning twice now? (It makes sense to me, but given the comment being removed, I wanted to ask)

Reply:
The removed comment is still true, but without Z3 (and the bug being fixed) the reasoning is not powerful enough to make use of the fact. Once Z3 gets merged in the comment will be back and the warning will be gone.